Styling improvements to openedX search through Find Courses page

marcotuts opened the pull request:
**Description**
This PR addresses some of the styling and UI changes that we'd like to incorporate into the course discovery feature currently in development. This is still a work in progress, and tasks / screenshots are included below. 
 
- - -
 
**JIRA Story** ([UX-2207](https://openedx.atlassian.net/browse/UX-2207))
**Confluence / Product Asset** ([Link to Confluence Docs](https://openedx.atlassian.net/wiki/display/SOL/OpenedX+Search+-+Requirements#OpenedXSearch-Requirements-Project3-CourseDiscoverySearchYellowINDISCOVERY))
**Sandbox URL** (Last Updated: 6/22) ([pr8296.m.sandbox.edx.org](http://pr8296.m.sandbox.edx.org)) 
**Dependencies** Main feature branch was merged to master. (https://github.com/edx/edx-platform/pull/7697).
 
- - -
**Must Have List**
- [x] tests passing
- [x] improve diff quality (currently is blocking merge.)
- [x] currently, search input element is fragile (especially with a styling change I added). It would take a bit more time to clean this up to work better on load and for mobile screen sizes.
- [x] remove common.py changes in this PR. 
- [x] review RTL (filter headers / txt and search header are broken currently) (MUST)
- [x] fix loading text which breaks styling, move to just on the outside left of the search area, or simply replace the search icon with the loading icon when loading. (Not sure what the best approach is here. cc'ed @dsego )

**Nice to Have / Should Have**
- [x] add "Viewing 126 courses" label when you have results. Currently we only have a label if there are zero results.

**Moved to JIRA Stories / Backlog**
- [x] transition for filtering sidebar to animate in (cc'ed @dsego) - ([SOL-959](https://openedx.atlassian.net/browse/SOL-959)) 
- [x] active filter should have different styling in sidebar as on edx.org (with clear / remove icon to also allow learners to unset a filter from the sidebar. ([SOL-959](https://openedx.atlassian.net/browse/SOL-959)) 
- [x] additional cleanup could be done for the styling / html work on search, if not in this PR, another story should be created. (this is more a styling organization and inheritance issue.) ([UX-2286](https://openedx.atlassian.net/browse/UX-2286))

- - -
**Screenshots** 
Find Courses Page:
![image](https://cloud.githubusercontent.com/assets/2023680/7737878/d652f084-ff1c-11e4-8c5b-523465b405e1.png)

Small changes to filters:
![image](https://cloud.githubusercontent.com/assets/2023680/7737883/e5ac39aa-ff1c-11e4-9a73-be46f694a07f.png)

Home page changes (including header for Find Courses link) :
![image](https://cloud.githubusercontent.com/assets/2023680/7737896/f8ee805e-ff1c-11e4-83bb-18f78cc0aedc.png)

- - -
**Reviewers**
- [x] @frrrances 
- [x] @mattdrayer 

- - -
**Product Verification**
- [ ] @pbaruah  

- - -
**FYI**
@talbs - The following PR updates the openedx course discovery work to move the find courses experience to its own page and update some of the styling to be closer to what we have on edx.org. As noted in the tasks above, there is still additional cleanup that should be explored. It might make sense to bundle that cleanup into a story that also incorporates the latest and greated PL work as well ;) Thanks!

marcotuts commented:
@dsego  It looks like this fails due to quality issues, instead of any specific tests. (https://build.testeng.edx.org/job/edx-platform-test-subset/84846/) I may get a chance to look at this tomorrow or thursday, but if you have previously run these quality tests before locally and know how to avoid any quality issues currently on this PR, your help would be great!

dsego commented:
@marco FYI, I rebased this branch.

marcotuts commented:
Awesome! Thanks @dsego.  It looks like we have passing tests here, so I'll update the sandbox and ping @frrrances and @pbaruah for reviews.

frrrances commented on lms/static/js/fixtures/discovery.html:
```diff
@@ -1,26 +1,28 @@
       <input class="discovery-input" placeholder="Search for a course" type="text"/><!-- removes spacing
       --><button type="submit" class="button postfix discovery-submit" aria-label="Search">
         <i class="icon fa fa-search" aria-hidden="true"></i>
+        <div aria-live="polite" aria-relevant="all">
```
Should this aria information be outside the form?

marcotuts replied:
I think the <aria-live="polite"> work was a pattern @dsego brought in from other a11y feedback / suggestions (?) @cptvitamin is this what we're expecting to do for the search input area or should we update any of this (or the surrounding markup?) thanks!

cptvitamin replied:
I have to say I've never seen anyone try to add an aria-live region as a child of button so i don't know what would happen. I would say that we should move this outside the button.  It's fine as a child of form, but should not be inside the button.

dsego replied:
This is file actually just the fixture for js tests, which should mimic lms/templates/courseware/courses.html .

frrrances commented on lms/static/js/fixtures/discovery.html:
```diff
@@ -1,26 +1,28 @@
 
-  <div id="filter-bar" class="filters hide-phone">
-  </div>
+    <section class="courses">
```
I think this should just be a div - there are already too many nested sections and div matches the other siblings, .discovery-form and .filter-bar

marcotuts replied:
I'm fine with changing this. Just wanted to double check with @cptvitamin if by chance the use of section tags here came out of an earlier a11y review. (Thanks for flagging this! )

cptvitamin replied:
thanks for checking @marcotuts.  It doesn't matter to me if its a div or a section.  according to the spec, a section should have a heading element and I don't see one here, so it should probably be a div.  However, what is going to be displayed in this list?  Would there be a benefit for a  screen user to be able to quickly navigate here?  If so, I would add a role="region" to the div and add an aria-label that describes the contents of the div.

marcotuts replied:
@cptvitamin  This is in fact containing a list of courses, so I added the region role and an aria label to both this and the text fixture. Thanks for the suggestions here!

frrrances commented on lms/static/sass/base/_variables.scss:
```diff
@@ -82,6 +82,27 @@ $gray-d2: shade($gray,40%); // #4c4c4c
 $gray-d3: shade($gray,60%); // #323232
 $gray-d4: shade($gray,80%); // #191919
 
+$cms-blue: rgb(0, 159, 230); // NOTE - eventually, this set of blues should be synced with CMS.  
```
While it sucks to have all our blues start with m- I don't think we should put in a whole new set. Can you use the other blue variables for now? Feel free to make a followup cleanup story for our backlog.

marcotuts replied:
This is all the same as the blues used in Studio, the only difference is the the base blue is called $cms-blue instead of calling it $blue and forcing all LMS blues over to CMS with this change. Using $blue-l2 on the LMS would be the same hex value as the $blue-l2 for CMS today. With this change, people can start using the lights / darks of this blue set instead of m-blue-* (which is the only option otherwise.) It felt wrong to keep adding more m-blue-* variables without doing some of the cleanup. We'd still need a cleanup task to remove all m-blue-* to use this syntax, as well as all update all uses of $blue, but this shouldn't impact any colors currently in the LMS. 

Does that make sense?

frrrances replied:
It does but the LMS blues have been adjusted to be more accessible - the base LMS blue is at the line for AA accessibility on light gray. Overall, I agree with your plan for cleanup and refactoring of the blues to be Studio-like, but I'm not sure this PR is the place for it.

marcotuts replied:
@frrrances That make sense.  I'll go through and update the colors here to go back to using m-blue-*. We can treat this as a separate cleanup task.

marcotuts replied:
I went ahead and backed out these blues to use the m-blue-* colors, and added back in the older blue-d1; etc used by other parts of the LMS.

frrrances commented on lms/static/sass/multicourse/_courses.scss:
```diff
@@ -257,160 +333,152 @@ $facet-background-color: #007db8;
-  .search-facets{
+// +Facets - Sidebar - Search
+// ====================
+.find-courses .search-facets {
```
why move these into a new .find-courses wrapper when there's one just above?

marcotuts replied:
merged with section above.

frrrances commented on lms/static/sass/multicourse/_courses.scss:
```diff
@@ -27,24 +45,14 @@ $facet-background-color: #007db8;
-        @include line-height(37.84);
-      }
-    }
+    @include padding(($baseline*3), ($baseline/2), 0, ($baseline/2));
```
I think this is too much top padding, $baseline*2 seems sufficient.

marcotuts replied:
updated.

frrrances commented on lms/static/sass/multicourse/_courses.scss:
```diff
@@ -257,160 +333,152 @@ $facet-background-color: #007db8;
-    }
+  .header-search-facets, .header-facet {
+    @extend %t-title6;
+    padding: ($baseline/2);
```
Might be nice to have an @extend %t-strong; here

frrrances commented on lms/static/sass/multicourse/_courses.scss:
```diff
@@ -257,160 +333,152 @@ $facet-background-color: #007db8;
+    @extend %t-title6;
+    padding: ($baseline/2);
+    margin: 0;
+    color: $facet-text-color;
```
Is this variable too specific? Should it be related to action colors or base text color?

marcotuts replied:
I just realized it is based on a hex value elsewhere, so it should definitely be updated. 

Should we drop in the base text color ($blue-somehtingorother) or does it make sense to keep abstracted out this way? For theming purposes I can see the benefit of having it defined up top in a single variable, but I know we're rebuilding that whole theming later so perhaps that consideration isn't important for now.

frrrances replied:
Yeah, definitely good to consider theming but it might be best to rely on the color theming at this point, since the rest isn't consistently applied. I'll leave it up to you though.

frrrances commented on lms/static/sass/multicourse/_courses.scss:
```diff
@@ -257,160 +333,152 @@ $facet-background-color: #007db8;
-      margin: 0 ($baseline/2) ($baseline/2) ($baseline/2);
-      color: $facet-text-color;
-      font-family: $sans-serif;
+  .search-facets-lists section {
```
Can you add a class to those children sections (maybe "search-facet-group"?) instead of styling the html element?

marcotuts replied:
@dsego - Would you mind pointing me toward where I might be able to add this class to achieve what @frrrances is suggesting? It looks to me like this is built up in javascript, as I can't seem to find the appropriate underscore template. Thanks!

frrrances commented on lms/static/sass/multicourse/_courses.scss:
```diff
@@ -257,160 +333,152 @@ $facet-background-color: #007db8;
-      color: $facet-text-color;
-      font-family: $sans-serif;
+  .search-facets-lists section {
+    border-top: 1px solid $courseware-button-border-color;
```
This doesn't seem like the right variable to be using. Switch to $gray-l4

marcotuts replied:
updated

frrrances commented on lms/static/sass/multicourse/_courses.scss:
```diff
@@ -257,160 +333,152 @@ $facet-background-color: #007db8;
+
+  .facet-option {
+    @include float(left);
+    @include line-height(18.92);
```
Use margin instead of line-height: 
```
margin: ($baseline/4) 0;
```

marcotuts replied:
updated.

frrrances commented on lms/static/sass/multicourse/_courses.scss:
```diff
@@ -257,160 +333,152 @@ $facet-background-color: #007db8;
-    section {
-      margin: 0;
-      padding: ($baseline/2) 0;
+    li {
```
Would prefer a class here.

dsego replied:
Uh, I think .facet-option is the button inside the li item.

frrrances commented on lms/static/sass/multicourse/_courses.scss:
```diff
@@ -257,160 +333,152 @@ $facet-background-color: #007db8;
-      @include float(left);
+    .count {
+      @include right($baseline*0.6);
+      @include float(right);
```
remove float. add positioning: 
```
position: absolute;
```

marcotuts replied:
updated. will review this on in RTL before merging to make sure it works as expected there.

frrrances commented on lms/static/sass/multicourse/_courses.scss:
```diff
@@ -257,160 +333,152 @@ $facet-background-color: #007db8;
+      @include float(right);
       @include box-sizing(border-box);
+      @include transition(all 0.2s ease-out);
       @include line-height(18.92);
```
remove line-height

frrrances commented on lms/static/sass/multicourse/_courses.scss:
```diff
@@ -257,160 +333,152 @@ $facet-background-color: #007db8;
-      overflow: hidden;
-      list-style: outside none none;
+    //STATE: hover
+    &:hover {
```
should this also be focus?

marcotuts replied:
updated to be both.

frrrances commented on lms/static/sass/multicourse/_courses.scss:
```diff
@@ -257,160 +333,152 @@ $facet-background-color: #007db8;
+// +All Other Styles
+// ====================
+.find-courses, .university-profile {
+  background: $course-profile-bg;
```
This seems like a funny variable to use here ... can you use a plain color one or will a change here make sense throughout the system?

marcotuts replied:
I've updated this to $gray-l5 directly.

frrrances commented on lms/static/sass/multicourse/_home.scss:
```diff
@@ -12,12 +12,10 @@ $course-search-input-height: ($button-size);
-    @include clearfix();
-    height: 460px;
     overflow: hidden;
     margin-top: $header_image_margin;
```
This margin needs to be removed  - it's -69px and messes up the layout in FF (but doesn't affect Chrome).

marcotuts replied:
removed.

frrrances commented on lms/static/sass/multicourse/_home.scss:
```diff
@@ -39,8 +37,7 @@ $course-search-input-height: ($button-size);
       position: relative;
-      border: 1px solid $border-color-3;
-      box-shadow: 0 4px 25px 0 rgba(0, 0, 0, 0.5);
+      box-shadow: 0 4px 5px 0 rgba(0, 0, 0, 0.5);
```
Nice shadow change!

frrrances commented on lms/static/sass/multicourse/_home.scss:
```diff
@@ -85,15 +82,14 @@ $course-search-input-height: ($button-size);
           height: $course-search-input-height;
           color: $black;
-          font-style: $sans-serif;
+          font-style: normal;
```
Can you also add this: 
```
font-weight: normal;
```

marcotuts replied:
+1. updated

frrrances commented on lms/static/sass/search/_search.scss:
```diff
@@ -123,7 +123,6 @@
 
 }
 
-
```
Clean up time! http://media.giphy.com/media/HIrYP4RI9DpLy/giphy.gif

frrrances commented on lms/static/sass/shared/_header.scss:
```diff
@@ -3,7 +3,7 @@
 header.global {
   @extend %ui-depth1;
-  border-bottom: 1px solid $m-gray;
+  border-bottom: 4px solid $courseware-border-bottom-color;
```
This seems like an edx change... Should it just be left as-is?

marcotuts replied:
the use of the courseware-border variable is intended to use the existing theming work. It does make the openedx header look closer to the edx, but I'm happy to revert the change.

frrrances commented on lms/static/sass/shared/_header.scss:
```diff
@@ -289,7 +289,7 @@ header.global {
         display:block;
         padding: ($baseline/4);
-        color: $lighter-base-font-color;
+        color: $link-color;
```
yay!

frrrances commented on lms/static/sass/shared/_header.scss:
```diff
@@ -302,7 +302,7 @@ header.global {
         a {
           text-decoration: none;
-          color: $link-color;
+          color: $lighter-base-font-color;
```
wait a sec...

marcotuts replied:
lol. resolving. :)

frrrances commented:
@marcotuts Thanks for doing this cleanup. I made a pass at the code and left you some comments. I also noticed the following visual and UX things: 

1. The search matching isn't great - it doesn't catch partial matches, including plurals, which seems like we're setting users up to not find things... Can we at a minimum default to showing all courses (along with the "no match" message at the top) if there is no match?

2. Clicking a card goes into the Course Info page inside the LMS instead of the Course About page... is that expected behavior?

3. The small tablet layout is a bit off - I think it would be better to have 2 columns for cards and one column for the header/search area. Also the top padding on the header should be adjusted along with the type size, maybe down to 20px.

![screen shot 2015-06-04 at 3 36 22 pm](https://cloud.githubusercontent.com/assets/4327102/7993783/a01d4364-0ad6-11e5-859b-e748ccf6cc70.png)
 
4. The homepage banner and courses filters are not looking quite right in FF. (I commented in the code on how to fix, but thought I would not it here as well)

![screen shot 2015-06-04 at 2 49 23 pm](https://cloud.githubusercontent.com/assets/4327102/7993703/f36659da-0ad5-11e5-81b1-edcc54508535.png)

![screen shot 2015-06-04 at 2 50 38 pm](https://cloud.githubusercontent.com/assets/4327102/7993687/e4dc47a8-0ad5-11e5-9d69-59a564efba30.png)

5. NIT: It's a bit weird that the header isn't responsive but the rest of the page is, but I get it if it's out of scope...

Let me know when you need another review.

dsego commented on lms/templates/courseware/courses.html:
```diff
@@ -52,33 +52,21 @@
+          <div aria-live="polite" aria-relevant="all">
+            <div id="loading-indicator" class="loading-spinner hidden">
+              <i class="icon fa fa-spinner fa-spin"></i>
+              <span class="sr"${_('Loading')}</span>
```
Missing angle bracket?

marcotuts replied:
resolved. thanks!

dsego commented on lms/static/sass/multicourse/_courses.scss:
```diff
@@ -257,160 +333,152 @@ $facet-background-color: #007db8;
+    width: 100%;
+    color: $facet-text-color;
+
+    &.collapse {
```
collapse does not belong here I think, it is a class added to .facet-list (to hide or show more options).

marcotuts replied:
ah thanks. must have moved that accidentally in the reorganization. moved into .facet-list.

cptvitamin commented on lms/static/js/fixtures/discovery.html:
```diff
@@ -1,26 +1,28 @@
         <i class="icon fa fa-search" aria-hidden="true"></i>
+        <div aria-live="polite" aria-relevant="all">
+          <div id="loading-indicator" class="loading-spinner hidden">
+          <i class="icon fa fa-spinner fa-spin"></i>
```
this icon is missing an aria-hidden attribute

marcotuts replied:
+1. updated.

marcotuts commented:
I've addressed all the feedback included above from @frrrances  and @cptvitamin  excepting the following items:

- [ ] Course items still link to /info page instead of /about page. @dsego could you advise on what change is necessary to fix this? 

- [ ] I didn't move the aria-polite area outside of the button, since the "Loading" text it is intending to hide is currently nested inside of this and moving it might have other implications. @dsego - Do you think this is ok to move outside? I'm not sure what the original intent of this aria-polite change is. It is possible also that @ahsan-ul-haq may have added this. (Referencing the conversation for @ahsan-ul-haq: https://github.com/edx/edx-platform/pull/8296#discussion-diff-31755290)

@frrrances (A few notes for other items you noted in your review)
* I've updated the tablet layout issue, and there is a separate follow-on story (in progress!) to make the header area responsive, so I've not addressed that note in this PR (https://github.com/edx/edx-platform/pull/8389)  
* As for the search matching, we have a story on the backlog to improve the index through stemming. As you noted, the matches today aren't great. (https://openedx.atlassian.net/browse/SOL-992)
* I have also created a story to add different logic for the empty results case, as this is something we should handle much more gracefully. https://openedx.atlassian.net/browse/SOL-991
cc'ed @pbaruah for the product backlog implications of the suggestions made in this PR.

Thanks everyone!

dsego commented:
The link to /info is baked into the template `result_item.underscore` [here](https://github.com/edx/edx-platform/blob/5b9ce69d88ea38d8fc76768da3a1c951a850c3a4/lms/templates/discovery/result_item.underscore#L2).

Moving the polite element should not cause problems. This is when it was added:
https://github.com/edx/edx-platform/pull/7697#discussion_r31267359

marcotuts commented:
Thanks @dsego for the pointed to the template. I've updated this and the corresponding test as well. @cptvitamin I'm spinning up a sandbox now for you to review the aria-polite work. 

@chrisndodge I wanted to also ask you to review this sandbox once it is up and running to make sure that things work as expected for white label after these changes. I was careful in the rebase to keep in the changes you added but it would still be good to have a quick review. 

(Sandbox available around noon EST)

cptvitamin commented on lms/static/js/fixtures/discovery.html:
```diff
@@ -1,26 +1,28 @@
+  <div id="discovery-form" class="wrapper-search-context">
+    <div id="discovery-message" class="search-status-label"></div>
+    <form class="wrapper-search-input">
       <input class="discovery-input" placeholder="Search for a course" type="text"/><!-- removes spacing
```
@marcotuts I just realized that this input has no ```<label>``` element associated with it.  Ideally, I like to see visible labels.  They persist (unlike placeholder text) which helps lowvision and those with a cognitive impairment.  However, sometimes that doesn't work with the UI.  Acceptable options are an offscreen label or aria-label if you can't make a visual label work.   Given the fact that we have placeholder text and an icon button, I'm OK with offscreen or aria-label here.

marcotuts replied:
Ah ok, there is currently an aria-label on the child <button> element. Should I move that aria-label to this input instead?

cptvitamin commented on lms/static/js/fixtures/discovery.html:
```diff
@@ -1,26 +1,28 @@
+    <div id="discovery-message" class="search-status-label"></div>
+    <form class="wrapper-search-input">
       <input class="discovery-input" placeholder="Search for a course" type="text"/><!-- removes spacing
       --><button type="submit" class="button postfix discovery-submit" aria-label="Search">
```
I'm not crazy about the approach taken for this button.  There is no accessible text here except for the aria-label.  Aria-label should only be used as a last resort.  It's not universally supported so it may not work in all environments. Ideally, we would have an SVG icon in here.  SVG supports accessible text so at least there would be something between the ```<button></button>``` elements.  Right now, we only have an icon font (which may not load, especially if the user has customized there browser fonts) and the "Loading" text which is not the primary label for this control.  I would prefer offscreen text here for the text value of this button.

marcotuts commented:
@smagoun, @pbaruah, @dsego  - I've run into issues getting these visual changes to the course discovery experience successfully rendering on a sandbox. After reviewing with @chrisndodge it seemed like it might be a configuration issue, but despite my best efforts I can't seem to get this to work even locally. By chance would any of the potential edx-search issues recently on master have anything to do with this?

More specifically, I can't get the sidebar to load in any facets / filters: 
![image](https://cloud.githubusercontent.com/assets/2023680/8190170/14bbe1ec-142f-11e5-93ad-e2958cc8fce5.png)

Sandbox link: http://pr8296.m.sandbox.edx.org/courses

Any support or suggestions on how to proceed would be helpful. Once this sandbox is ready I can ask Frances, Davorin, and Mark to do a final review. 

Thanks!

chrisndodge commented:
@marcotuts looks like we have failing unit tests. Might be easiest to try to debug by fixing the unit tests. I'll take a look and see what's happening, although I don't have a lot of experience in this area of the code.

chrisndodge commented:
@marcotuts AAARGGG! Now I need to update the python unit test to account for the change from section -> div :-( I'll do so later tonight.

marcotuts commented:
@pbaruah, @dsego, @frrrances  - I've updated the sandbox with the latest, you can now do your second / final reviews, or in the case or partha - product acceptance / review. http://pr8296.m.sandbox.edx.org/courses

Thanks everyone!

marcotuts commented:
NOTE: I have not configured the names of the search sidebar filters, but this is a configuration that is enabled behind the scenes. The effect of this is that words like "language" and "org" could be capitalized, and values such as the language codes shown would be "English" instead of "en" 
https://github.com/edx/edx-platform/blob/e75a600048a7dc4ffb15cbe72426965236427b3c/lms/envs/devstack.py#L137

marcotuts commented on lms/static/sass/views/_homepage.scss:
```diff
@@ -16,7 +16,7 @@ $learn-more-horizontal-position: calc(50% - 100px); // calculate the left positi
 
       .courses-listing-item {
-        margin: ($baseline*0.75) 0 ($baseline*1.5) 0;
+        margin: 0 0 ($baseline*1.5) 0;
```
NIT: this is not being caught by specificity rules, margin-top remains which means the top row of courses isn't aligned to the top of the sidebar.

frrrances commented:
Considering the followup stories, this looks good to me. :+1:

dsego commented:
@marcotuts I took the liberty of rebasing the branch onto master.

mattdrayer commented on lms/templates/navigation.html:
```diff
@@ -124,15 +124,21 @@ <h2 class="course-header"><span class="provider">${course.display_org_with_defau
         % endif
       </%block>
       % if not settings.FEATURES['DISABLE_LOGIN_BUTTON']:
+          % if settings.FEATURES.get('ENABLE_COURSE_DISCOVERY'):
```
Rather than checking settings directly here, it would be better to create a context parameter such as 'course_discovery_enabled' and then determine the status of this value in the view function

marcotuts replied:
@mattdrayer Is it fair to punt these changes over to another story / PR? I think all of our feature flag checks are done as is currently. I'm happy to draft that story if that works for you. 

Additionally, are you the second reviewer on this or were you just passing through providing useful feedback? I wanted to check in since this PR needs a second reviewer. cc'ed @pbaruah. 

@dsego was originally going to be a reviewer but as he contributed to the PR it might make sense for someone else to do a review of that work.

mattdrayer replied:
I'm fine punting the changes, but please do be sure to capture them in a backlog somewhere.  @pbaruah asked if I could do a review of this PR, but I can't really comment on the Sass changes, so you may want to find someone else -- perhaps another guy from EE?

marcotuts replied:
Ah ok thanks for the context and for flagging this. We have had Frances review the SASS changes here so if your review was for the rest of the PR that seems good to me. 

Here is the story I created with the details you described above. https://openedx.atlassian.net/browse/SOL-1043

Thanks!

mattdrayer commented on lms/templates/navigation.html:
```diff
@@ -124,15 +124,21 @@ <h2 class="course-header"><span class="provider">${course.display_org_with_defau
+            <a class="cta cta-discovery" href="/courses">${_("Find Courses")}</a>
+          </li>
+          %endif
           % if course and settings.FEATURES.get('RESTRICT_ENROLL_BY_REG_METHOD') and course.enrollment_domain:
```
As above, it would be better to move this logic into the view code and set a context parameter like 'show_registration_link" to True/False

mattdrayer commented:
:+1:  for me after discussing follow-up changes w/ Marco

marcotuts commented:
Thanks @mattdrayer, @dsego, @frrrances for all the support on this. I'll connect with @pbaruah for product verification / acceptance and then we'll be ready to merge.

pbaruah commented:
:+1:

catong commented:
@srpearce FYI for any doc impacts.
